src/java/org/apache/cassandra/index/sai/plan/QueryView.java

```diff
@@ -199,33 +219,5 @@ else if (MonotonicClock.approxTime.now() - failingSince > TimeUnit.MILLISECONDS.
                 }
             }
         }
-
-        /**
-         * Get the index
-         */
-        private Set<SSTableIndex> getIndexesForExpression(IndexContext indexContext)
-        {
-            if (!indexContext.isIndexed())
-                throw new IllegalArgumentException("Expression is not indexed");
-
-            // Get all the indexes in the range.
-            return indexContext.getView().getIndexes().stream().filter(this::indexInRange).collect(Collectors.toSet());
-        }
-
-        // I've removed the concept of "most selective index" since we don't actually have per-sstable
-        // statistics on that; it looks like it was only used to check bounds overlap, so computing
-        // an actual global bounds should be an improvement.  But computing global bounds as an intersection
-        // of individual bounds is messy because you can end up with more than one range.
-        private boolean indexInRange(SSTableIndex index)
-        {
-            SSTableReader sstable = index.getSSTable();
-            if (range instanceof Bounds && range.left.equals(range.right) && (!range.left.isMinimum()) && range.left instanceof DecoratedKey)
-            {
-                if (!sstable.getBloomFilter().isPresent((DecoratedKey)range.left))
```

Comment:
I remember that this was added as an optimization. Perhaps we could still filter out indexes if they are known not to have the specific partition? This would be helpful in cases where the table relies on partitioning vectors into groups and then always restricts by partition.

Reply:
Makes sense. I'll add that back in the proper place.

Reply:
Great. And fwiw, I think `testIndexQueryWithPartitionKey` was failing because this was removed, so you would have found it either way.